state/factory/workingset.go

```diff
@@ -346,6 +345,11 @@ func (ws *workingSet) State(s interface{}, opts ...protocol.StateOption) (uint64
 	if cfg.Keys != nil {
 		return 0, errors.Wrap(ErrNotSupported, "Read state with keys option has not been implemented yet")
 	}
+	if ws.parent != nil {
```

Comment:
Should first query from the current ws, and if not found, then query from the parent

Reply:
good point, but this is a little complicated, for example, assume
1. in the DB, `state(k) = 100`
2. in the parent workingset `state(k) = 98` (there's a transfer of 2 token in pending block)
in this case, we should honor the value in parent workingset
assume there's another transfer of 3 token in current block, then `state(k) = 95` now, and things get more complicated: for next read, should honor the value in current working
i think the return value of `State/Read` should indicate if the value is from workingset cache, and honor the value in cache over that from DB

Reply:
maybe we can replace db store with parent